### PR TITLE
quotes_handler: Check quotes in unit tests

### PR DESCRIPTION
**Pull request by ueno**
This is a follow-up of #265 to add proper testing of generated quotes using the same logic in tpm2_checkquote.

### CODE REVIEW

File: src/tpm.rs
Comment (lkatalin) [inline src/tpm.rs:199]:
```suggestion
    // TPML_PCR_SELECTION should be [u8; 132]
    let pcrsel_vec: [u8; std::mem::size_of::<TPML_PCR_SELECTION>()] =
```
These were extremely tricky and Patrick had to hardcode them for some reason; the sizes were not what we expected. If everything still seems to run okay (I will check) then I think this change is fine, but I would want to record these values in a comment in case something goes wrong.

File: src/tpm.rs
Comment (lkatalin) [inline src/tpm.rs:203]:
```suggestion
    // TPML_DIGEST should be [u8; 532]
    let digest_vec: [u8; std::mem::size_of::<TPML_DIGEST>()] =
```

File: src/tpm.rs
Comment (lkatalin) [inline src/tpm.rs:204]: This one seems straightforward and should be changed.

File: src/tpm.rs
Comment (lkatalin) [inline src/tpm.rs:211]: I like this comment, thanks.

File: src/tpm.rs
Comment (lkatalin) [reply src/tpm.rs:203]: That's awesome! I didn't know about that crate. Thanks, @ueno !

**Comment by lkatalin** [comment]
@ueno I'm not sure where this left off. It seems like it was ready to merge, so my apologies if I dropped it. Would you be able to rebase and I can test it to make sure?

**Comment by ueno** [comment]
@lkatalin no worries and sorry for the delay from my side; rebased

**Comment by lkatalin** [comment]
@ueno Looks like it needs another rebase / resolve conflicts after merging #268

**Comment by ueno** [comment]
@lkatalin should be ready now

**Comment by lkatalin** [comment]
Thank you, @ueno , this is incredibly helpful to have in the code.

**Review by lkatalin — APPROVED** [review]

**Pull request merged**
